frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -978,14 +984,24 @@ export const dashboardLogic = kea<dashboardLogicType>([
                 return
             }
 
-            let breakpointTriggered = false
+            let cancelled = false
             actions.setRefreshStatuses(
                 insights.map((item) => item.short_id),
                 true
             )
 
+            // If a query is in progress, kill that query
```

Comment:
Non-blocking note: It feels like there's a missing abstraction here that would make this code a lot less unwieldy. Missing kea feature?